From: Michael Carroll <mjcarroll.oss@gmail.com>
Date: Tue, 5 May 2026 02:26:44 -0500
Subject: [PATCH] Fix Pytest 8/9 compatibility in hooks

Backport of upstream commit 06308316b46a734188a6688603763185d2b3de54.

Trimmed to the launch_testing package and adjusted for the package source layout.

Signed-off-by: Michael Carroll <mjcarroll.oss@gmail.com>
Signed-off-by: Xu Colin <colinxu2020@gmail.com>
---
 launch_testing/pytest/hooks.py             | 18 +++++++++++++-----
 launch_testing/pytest/hookspecs.py         |  2 +-
 2 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/launch_testing/pytest/hooks.py b/launch_testing/pytest/hooks.py
index 174c4c476..3726d9570 100644
--- a/launch_testing/pytest/hooks.py
+++ b/launch_testing/pytest/hooks.py
@@ -189,12 +189,20 @@ def find_launch_test_entrypoint(path):
         return None
 
 
-def pytest_pycollect_makemodule(path, parent):
+if _pytest_version_ge(8):
+    def pytest_pycollect_makemodule(module_path, parent):
+        return _pytest_pycollect_makemodule(module_path, parent)
+else:
+    def pytest_pycollect_makemodule(path, parent):
+        return _pytest_pycollect_makemodule(path, parent)
+
+
+def _pytest_pycollect_makemodule(path, parent):
     entrypoint = find_launch_test_entrypoint(path)
     if entrypoint is not None:
         ihook = parent.session.gethookproxy(path)
         module = ihook.pytest_launch_collect_makemodule(
-            path=path, parent=parent, entrypoint=entrypoint
+            module_path=path, parent=parent, entrypoint=entrypoint
         )
         if module is not None:
             return module
@@ -216,14 +224,14 @@ def pytest_pycollect_makemodule(path, parent):
 
 
 @pytest.hookimpl(trylast=True)
-def pytest_launch_collect_makemodule(path, parent, entrypoint):
+def pytest_launch_collect_makemodule(module_path, parent, entrypoint):
     marks = getattr(entrypoint, 'pytestmark', [])
     if marks and any(m.name == 'launch_test' for m in marks):
         if _pytest_version_ge(7):
-            path = pathlib.Path(path)
+            path = pathlib.Path(module_path)
             module = LaunchTestModule.from_parent(parent=parent, path=path)
         else:
-            module = LaunchTestModule.from_parent(parent=parent, fspath=path)
+            module = LaunchTestModule.from_parent(parent=parent, fspath=module_path)
         for mark in marks:
             decorator = getattr(pytest.mark, mark.name)
             decorator = decorator.with_args(*mark.args, **mark.kwargs)
diff --git a/launch_testing/pytest/hookspecs.py b/launch_testing/pytest/hookspecs.py
index 89249d5ea..52912415e 100644
--- a/launch_testing/pytest/hookspecs.py
+++ b/launch_testing/pytest/hookspecs.py
@@ -16,6 +16,6 @@
 
 
 @pytest.hookspec(firstresult=True)
-def pytest_launch_collect_makemodule(path, parent, entrypoint):
+def pytest_launch_collect_makemodule(module_path, parent, entrypoint):
     """Make launch test module appropriate for the found test entrypoint."""
     pass
